app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt

```diff
@@ -446,8 +488,25 @@ class SettingsFragment : PreferenceFragmentCompat() {
         }
     }
 
+    @VisibleForTesting
+    internal fun setupAmoCollectionOverridePreference(settings: Settings) {
+        val preferenceAmoCollectionOverride =
+            findPreference<Preference>(getPreferenceKey(R.string.pref_key_override_amo_collection))
+
+        val show = (Config.channel.isNightlyOrDebug && (
+            settings.overrideAmoUser.isNotEmpty() ||
+            settings.overrideAmoCollection.isNotEmpty() ||
```

Comment:
nit: Maybe we could have this inside of `Settings` as a property as it's been also use in other places

Reply:
Updated and combined in a helper function.